backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/FunctionReferenceLowering.kt

```diff
@@ -282,7 +282,8 @@ internal class FunctionReferenceLowering(val context: Context): FileLoweringPass
                 else
                     kFunctionImplSymbol.typeWith(referencedFunction.returnType)
                 functionClass = (if (isKFunction) symbols.kFunctionN(numberOfParameters) else symbols.functionN(numberOfParameters)).owner
-                superTypes += functionClass.typeWith(functionParameterTypes + referencedFunction.returnType)
+                if (samSuperType == null)
```

Comment:
invoke method generation below probably needs an update too.
I guess current version generates a bogus `overriddenSymbols` entry now, which is inconsistent with actual super types.